### PR TITLE
add PasswordApi

### LINKED ISSUE #3175: Create PasswordHelper class
replace UserUtil::Password functions with password helper service or find alternative.

### PR DESCRIPTION
**Pull request by craigh**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no
| Deprecations?     | yes (old Utils)
| Fixed tickets     | #3175
| Refs tickets      | -
| License           | MIT
| Changelog updated | yes

## Description
add PasswordApi. fixes #3175

## Todos
- [x] Tests
- [x] Documentation
- [x] Changelog

### CODE REVIEW

**Comment by craigh** [comment]
none of the available methods are considered secure any longer. This is one reason to completely get out of the authentication business entirely (use OAuth instead). We need to upgrade all path words to php 5.5+ methods which use `mcrypt()` I think. There is another ticket on this.

**Comment by craigh** [comment]
refs https://github.com/zikula/core/issues/2842

**Comment by Guite** [comment]
> completely get out of the authentication business entirely

is this possible when using the internal user/password login only?

**Comment by craigh** [comment]
> is this possible

no

File: src/system/ZAuthModule/Controller/AccountController.php [hunk 1]
```diff
@@ -373,7 +375,7 @@ public function changePasswordAction(Request $request)
         $form->handleRequest($request);
         if ($form->isSubmitted() && $form->isValid()) {
             $data = $form->getData();
-            $mapping->setPass(\UserUtil::getHashedPassword($data['pass']));
+            $mapping->setPass($this->container->get('zikula_zauth_module.api.password')->getHashedPassword($data['pass']));
```
Comment (Guite) [inline src/system/ZAuthModule/Controller/AccountController.php]: Sometimes you use `$this->get`, sometimes `$this->container->get`. Please use `$this->get` only in controllers for consistency.

File: src/system/ZAuthModule/Tests/Api/PasswordApiTest.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        $this->assertEquals(2, substr_count($hashedPass, '$'));
+
+        $hashedPass = $this->api->getHashedPassword('H4ppy81rthd$y', 1); // 1 = md5
+        $this->assertEquals(40, strlen($hashedPass));
+        $this->assertRegExp(';[0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ~@#$%^*()_+-={}|\][];', $hashedPass);
+        $this->assertEquals(2, substr_count($hashedPass, '$'));
+
+        $hashedPass = $this->api->getHashedPassword('mybirthdayplusabunchofchanracters%&*&^53', 5); // 5 = sha1
+        $this->assertEquals(48, strlen($hashedPass));
+        $this->assertRegExp(';[0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ~@#$%^*()_+-={}|\][];', $hashedPass);
```
Comment (craigh) [inline src/system/ZAuthModule/Tests/Api/PasswordApiTest.php]: it is only used once?

**Pull request merged**